app/forms/SubscriptionsForm.scala

```diff
@@ -26,12 +27,13 @@ object SubscriptionsForm {
   private val addressMaxLength = 255
   private val emailMaxLength = 240
 
+  def unapplyUkAddress(a: Address) = Some((a.lineOne, a.lineTwo, a.town, a.postCode))
   val addressDataMapping = mapping(
     "address1" -> text(0, addressMaxLength),
     "address2" -> text(0, addressMaxLength),
     "town" -> text(0, addressMaxLength),
     "postcode" -> text(0, addressMaxLength)
-  )(AddressData.apply)(AddressData.unapply)
+  )(Address.apply(_:String, _:String, _: String, "United Kingdom", _: String, Countries.UK))(unapplyUkAddress)
```

Comment:
The string "United Kingdom" is wrong in this context, it maps the form data to an Address that has "United Kingdom" as a countyOrState property.